shpc/main/modules/templates/singularity.lua

```diff
@@ -72,10 +73,10 @@ if (myShellName() == "bash") then
 end{% endif %}
 
 -- A customizable exec function
-set_shell_function("{|module_name|}-exec", execCmd .. containerPath .. " ${SINGULARITY_COMMAND_ARGS}  $@",  execCmd .. containerPath .. " ${SINGULARITY_COMMAND_ARGS} $*")
+set_shell_function("{|module_name|}-exec", execCmd .. containerPath .. " ${SINGULARITY_COMMAND_ARGS}  $@",  execCmd .. containerPath .. " ${SINGULARITY_COMMAND_ARGS}")
```

Comment:
```suggestion
set_shell_function("{|module_name|}-exec", execCmd .. containerPath .. " ${SINGULARITY_COMMAND_ARGS}  $@",  execCmd .. containerPath)
```

Reply:
I'm thinking not - I removed it since you originally removed `$*` (but let me know if this is wrong and I should add it back).

Reply:
The `$@` is needed in shell function as functions are not expanded like aliases are.

The `set_shell_function` of Lmod use its second argument to define a shell function if current shell is an sh shell, third argument is the code to set a shell alias if current shell is a csh shell.

Reply:
Gotcha, added it back. And the csh doesn't need to accept args?

Reply:
For csh an alias is set and as any alias it is expanded in place so the remaining word after this alias name are de-facto its arguments